### ISSUE TITLE
Ability to have clients, and/or projects

### ISSUE THREAD

**Issue by BrendanThompson** [NullDesk/ticketdesk#26]
Would be ideal to be able to have the concept of _clients_ that a ticket can be associated with, as well as projects

**Comment by soaTf** [comment]
@StephenRedd Hi Stephen - we are evaluating your TicketDesk project (I already created a german translation). But we need 2 more roles in our helpdesk environment:
- "End User"/"Client": able to create and view his tickets and the tickets of other users assigned to the Client (multiple contact persons for a client, f.e. *@customerdomain.com)
- Optional "Partner": Able to create and view tickets in behalf of his assigned customers 
Therfore I would like to ask you about your TD 3 development plans.
thank you in advance
tobias

**Comment by StephenRedd** [comment]
@soaTf What you are describing is really a customer support system -- a multi-tenant system serving internal and external users. While the core of those systems do the same thing, there are significant behavioral differences between external support and internal support systems. 

TD is primarily an internal facing, and single-tenant, system -- it assumes high-trust for all users, and one of its design intentions is to avoid needing to understand complex (HR) organizational relationships. External systems usually need some aspects of CRM to identify the relationship between external users, and their organizations to your own, as well as complex knowledge of your own internal HR organization too. 

So, long story short -- TicketDesk, out of the box, is unlikely to meet your needs. It could serve as a nice starting point for such a system, since the core concepts (tickets, events, notifications, etc)  are the same. But the rest would be up to you to code around that core.

TD 3 is still in design, and thus largely still theoretical at this point, it will not be done within the next few months. It will be much more flexible than TD 2, and of course use much newer technologies. But overall, I had not planned to expand it into a full-featured customer-facing support system though... honestly, that's the sort of massive investment in time and effort that I usually save for paid-projects.

**Comment by soaTf** [comment]
@StephenRedd Thank you very much for your feedback - we will consider using TD as a starting point :-)
